### DISCUSSION TITLE
Instead of Nvidia, can we use OpenVino for Intel Processors?

### DISCUSSION THREAD

**Discussion by Dvalin21** [exadel-inc/CompreFace#1055] [Q&A] [answered]
Same as the title, and if so, please share how.  Thanks.

**Comment by pospielov** [comment, marked as answer]
CompreFace doesn't support OpenVino now. 
The problem is that to add support for CUDA, all we need is to install `tensorflow-gpu` library instead of `tensorflow`. 
OpenVino is another framework. I played around with it a couple of years ago and it was very raw without proper documentation.
I hope Intel will add somehow support for OpenVino to popular libraries like tensorflow and pytorch

**Reply by Dvalin21** [reply]
Thanks!

**Reply by PovilasID** [reply]
[Frigate NVR](https://github.com/blakeblackshear/frigate) now has an option to run on OpenVINO and it's awesome (low power consumption and fast). I would love to be able to reduce your power consumption or not to have add some crappy GPU to the system just for to run this...

It looks like OpenVINO now has `tensorflow` integration https://docs.openvino.ai/latest/ovtf_integration.html and model conversion option for `pytorch` https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_PyTorch.html

Though docs for both look... limited. Not sure if that is what you meant "without proper documentation" but if you could take a second look would be very nice

**Reply by pospielov** [reply]
Here is a trick. If you want to use OpenVino, you can use lots of converters (and yes, when I tried it several years ago, there did not exist).
But it still means we need to create a separate build for  `compreface-core` that uses OpenVino libraries, and put there converted models. 
The ideal situation for us will be if we continue to use tensorflow without changing the code, as it works for nvidia GPUs now. 
We discovered that [ONNXRuntime](https://onnxruntime.ai/docs/execution-providers/) gives such an ability, but we will still need to migrate `compreface-core` to it.

**Reply by Dvalin21** [reply]
I look forward to the updates
